--- a/wpa_supplicant/systemd/wpa_supplicant.service.in
+++ b/wpa_supplicant/systemd/wpa_supplicant.service.in
@@ -6,7 +6,7 @@ Wants=network.target
 [Service]
 Type=dbus
 BusName=fi.w1.wpa_supplicant1
-ExecStart=@BINDIR@/sdcsupp -u
+ExecStart=@BINDIR@/sdcsupp -u -O /run/wpa_supplicant
 
 [Install]
 WantedBy=multi-user.target
--- a/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service.in
+++ b/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service.in
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=fi.w1.wpa_supplicant1
-Exec=@BINDIR@/sdcsupp -u
+Exec=@BINDIR@/sdcsupp -u -O /run/wpa_supplicant
 User=root
 SystemdService=wpa_supplicant.service
