diff --git a/Makefile.am b/Makefile.am
index 7cc50332d5..244d3d8ae5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1864,6 +1864,7 @@ src/libnm-client-impl/libnm.typelib: src/libnm-client-impl/libnm.gir
 
 INTROSPECTION_GIRS += src/libnm-client-impl/NM-1.0.gir
 
+if BUILD_DOCS
 libnm_noinst_data = \
 	src/nmcli/generate-docs-nm-settings-nmcli.xml \
 	man/nm-settings-docs-dbus.xml \
@@ -1877,6 +1878,7 @@ libnm_noinst_data = \
 	$(NULL)
 
 noinst_DATA += $(libnm_noinst_data)
+endif
 
 EXTRA_DIST += src/nmcli/generate-docs-nm-settings-nmcli.xml.in
 
@@ -1893,6 +1895,7 @@ src/nmcli/generate-docs-nm-settings-nmcli.xml: src/nmcli/generate-docs-nm-settin
 check-local-generate-docs-nm-settings-nmcli:
 endif
 
+if BUILD_DOCS
 libnm_docs_sources = $(src_libnm_core_impl_lib_c_settings_real)
 
 src/libnm-client-impl/nm-property-infos-%.xml: tools/generate-docs-nm-property-infos.py $(libnm_docs_sources)
@@ -1919,6 +1922,7 @@ man/nm-settings-docs-%.xml: src/libnm-client-impl/nm-property-infos-%.xml src/li
 
 EXTRA_DIST += $(libnm_noinst_data)
 DISTCLEANFILES += $(libnm_noinst_data)
+endif
 
 endif
 
diff --git a/configure.ac b/configure.ac
index 46bfe7a027..ba1b249de5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1243,7 +1243,7 @@ else
 fi
 AC_SUBST(NM_LOG_COMPILER, 'LOG_COMPILER = "$(top_srcdir)/tools/run-nm-test.sh" --called-from-make "$(abs_top_builddir)" "$(LIBTOOL)" "$(with_valgrind)" "'"$with_valgrind_suppressions"'" --launch-dbus=auto')
 
-AX_PTHREAD([,AC_MSG_ERROR([Threads are required for the NetworkManager tests])])
+AX_PTHREAD([], [AC_MSG_ERROR([Threads are required for the NetworkManager tests])])
 
 if test -n "$PYTHON" ; then
     AM_PATH_PYTHON([], [], [])
@@ -1261,22 +1261,22 @@ GTK_DOC_CHECK(1.0)
 # check if we can build setting property documentation
 build_docs=no
 if test -n "$INTROSPECTION_MAKEFILE"; then
-	# If g-i is installed we know we have python, but we might not have pygobject
-	if ! "$PYTHON" -c 'from gi.repository import GObject' > /dev/null 2>&1; then
-		AC_MSG_ERROR(["--enable-introspection aims to build the settings documentation. This requires GObject introspection for python (pygobject)])
-	fi
-
-	AC_PATH_PROG(PERL, perl)
-	if test -z "$PERL"; then
-		AC_MSG_ERROR([--enable-introspection requires perl])
-	fi
-	AC_PATH_PROG(XSLTPROC, xsltproc)
-	if test -z "$XSLTPROC"; then
-		AC_MSG_ERROR([--enable-introspection requires xsltproc])
-	fi
-
 	have_introspection=yes
 	if test "$enable_gtk_doc" = "yes"; then
+		# If g-i is installed we know we have python, but we might not have pygobject
+		if ! "$PYTHON" -c 'from gi.repository import GObject' > /dev/null 2>&1; then
+			AC_MSG_ERROR(["--enable-gtk-doc aims to build the settings documentation. This requires GObject introspection for python (pygobject)])
+		fi
+
+		AC_PATH_PROG(PERL, perl)
+		if test -z "$PERL"; then
+			AC_MSG_ERROR([--enable-gtk-doc requires perl])
+		fi
+		AC_PATH_PROG(XSLTPROC, xsltproc)
+		if test -z "$XSLTPROC"; then
+			AC_MSG_ERROR([--enable-gtk-doc requires xsltproc])
+		fi
+
 		build_docs=yes
 	fi
 else
diff --git a/meson.build b/meson.build
index 37c2fe1c2c..70f9a16ce9 100644
--- a/meson.build
+++ b/meson.build
@@ -895,22 +895,22 @@ data_conf.set('sysconfdir',                              nm_sysconfdir)
 '''
 build_docs=no
 if test -n "$INTROSPECTION_MAKEFILE"; then
-  # If g-i is installed we know we have python, but we might not have pygobject
-  if ! "$PYTHON" -c 'from gi.repository import GObject' >& /dev/null; then
-    AC_MSG_ERROR(["--enable-introspection aims to build the settings documentation. This requires GObject introspection for python (pygobject)])
-  fi
-
-  AC_PATH_PROG(PERL, perl)
-  if test -z "$PERL"; then
-    AC_MSG_ERROR([--enable-introspection requires perl])
-  fi
-  AC_PATH_PROG(XSLTPROC, xsltproc)
-  if test -z "$XSLTPROC"; then
-    AC_MSG_ERROR([--enable-introspection requires xsltproc])
-  fi
-
   have_introspection=yes
   if test "$enable_gtk_doc" = "yes"; then
+    # If g-i is installed we know we have python, but we might not have pygobject
+    if ! "$PYTHON" -c 'from gi.repository import GObject' >& /dev/null; then
+      AC_MSG_ERROR(["--enable-gtk-doc aims to build the settings documentation. This requires GObject introspection for python (pygobject)])
+    fi
+
+    AC_PATH_PROG(PERL, perl)
+    if test -z "$PERL"; then
+      AC_MSG_ERROR([--enable-gtk-doc requires perl])
+    fi
+    AC_PATH_PROG(XSLTPROC, xsltproc)
+    if test -z "$XSLTPROC"; then
+      AC_MSG_ERROR([--enable-gtk-doc requires xsltproc])
+    fi
+
     build_docs=yes
   fi
 else
